lib/ddtrace/contrib/rails/patcher.rb

```diff
@@ -0,0 +1,55 @@
+module Datadog
+  module Contrib
+    module Rails
+      # Patcher
+      module Patcher
+        include Base
+        register_as :rails, auto_patch: true
+
+        option :enabled, default: true
```

Comment:
We've a different logic here: https://github.com/DataDog/dd-trace-rb/pull/226/files#diff-acecea54cdb8458166cefacb04d7a9baR26

Should we do the same? Because in Rails we're [disabling / enabling the tracer](https://github.com/DataDog/dd-trace-rb/blob/996453ea028295eeafe65128e1cfda8fc6e4b7e5/lib/ddtrace/contrib/rails/framework.rb#L41) when the Railtie is executed, but not when the initializer is called. 

It means that you set the `Tracer.enabled <- False` but the tracer can still send traces until Rails isn't fully initialized. Am I missing something?

Reply:
For the future, it would be great accessing directly to the tracer so that tracer specific configurations aren't part of integrations. That way as we've discussed before, you configure the Tracer AND Rails.

Reply:
I agree with everything you pointed out, but I'd keep things as they're now to avoid changing too many things at once. Right now we're maintaining exactly the same behavior as before. The next step would be getting rid of the `Framework.configure` method entirely, by using the pattern you described (using the custom setters provided in the new API)

Reply:
👍